src/main/java/org/folio/circulation/domain/notice/PatronNoticeService.java

```diff
@@ -19,36 +19,43 @@
 import org.apache.commons.lang3.builder.EqualsBuilder;
 import org.apache.commons.lang3.builder.HashCodeBuilder;
 import org.apache.commons.lang3.tuple.Pair;
+import org.folio.circulation.domain.representations.logs.NoticeLogContext;
 import org.folio.circulation.domain.notice.schedule.ScheduledNoticeConfig;
 import org.folio.circulation.infrastructure.storage.notices.PatronNoticePolicyRepository;
 import org.folio.circulation.rules.AppliedRuleConditions;
+import org.folio.circulation.services.EventPublisher;
 import org.folio.circulation.support.Clients;
 import org.folio.circulation.support.CollectionResourceClient;
 import org.folio.circulation.support.results.Result;
 import org.folio.circulation.support.http.client.ResponseInterpreter;
 
 import io.vertx.core.json.JsonObject;
+import org.joda.time.DateTime;
 
 public class PatronNoticeService {
-  public static PatronNoticeService using(Clients clients) {
-    return new PatronNoticeService(new PatronNoticePolicyRepository(clients), clients);
+  public static PatronNoticeService using(Clients clients, EventPublisher eventPublisher) {
```

Comment:
Am I correct understand that every notice should be logged?

Reply:
Yes, all sent notices